data/src/main/java/com/linkedin/data/schema/SchemaToPdlEncoder.java

```diff
@@ -719,6 +719,7 @@ private void writeDocAndProperties(RecordDataSchema.Field field) throws IOExcept
    * (1) The type is outside the root namespace of the document.
    * (2) The type is declared outside the document (i.e. not inlined in this document).
    * (3) The type's name does not conflict with name of an Inlined type.
+   * (4) The type's name does not conflict with another type that can use simple reference.
```

Comment:
I'd say `Importing the type wouldn't force some intra-namespace reference to become fully-qualified.`

Reply:
fixed